### PR TITLE
124-create component description page structure

### PR DESCRIPTION
**Pull request by dolinetouko**

### CODE REVIEW

**Review by Tayebsed93 — APPROVED** [review]

**Review by ludovic35 — CHANGES_REQUESTED** [review]

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: app/assets/il_radiobutton.png
Comment (paulinea) [inline app/assets/il_radiobutton.png:1]: Remove this file cause the equivalent exists in `.svg`

File: app/lib/ui/components/component_list.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  OdsComponent(
+      "Cards",
+      'assets/il_cards.svg',
+      "Cards are important components that can be used to organise and present a number of different types of related information.",
+      [
+        Variant("Small card", "OdsSmallCard", ComponentCardSmall()),
+        Variant("Vertical image first card", "OdsVerticalImageFirstCard",
+            Text("Coming Soon")),
+        Variant("Vertical header first card", "OdsVerticalHeaderFirstCard",
+            Text("Coming Soon")),
```
Comment (paulinea) [inline app/lib/ui/components/component_list.dart:24]: I think we have to remove this variants cause they don't exist for now in the Flutter library and we have dead links.

File: app/lib/ui/components/template/component_detail_screen.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      leading: Icon(Icons.play_circle_outline),
+      title: Text(variant.title),
+      subtitle: Text(variant.technicalName),
+      onTap: () {
+        Get.to(variant.screen);
+      },
```
Comment (paulinea) [inline app/lib/ui/components/template/component_detail_screen.dart:61]: The play icon should be centered vertically in the tile

File: app/lib/ui/components/template/component.dart
Comment (paulinea) [inline app/lib/ui/components/template/component.dart:1]: We discussed about directories with Ludovic and we decided that `component.dart`, `component_detail_screen.dart` and `components_screen.dart` should be moved out of the `template` directory. After that, the `template` directory should be removed.

File: app/lib/ui/components/template/component_detail_screen.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            displayImage(
+              component.imageResourceName,
+              MediaQuery.of(context).size.width,
+            ),
+            Padding(
+              padding: const EdgeInsets.only(
+                  top: spacingM,
+                  right: spacingM,
+                  left: spacingM,
+                  bottom: spacingM),
```
Comment (paulinea) [inline app/lib/ui/components/template/component_detail_screen.dart:30]: When the padding is the same everywhere, prefer the use of `EdgeInsets.all(spacingM)`

**Review by paulinea — CHANGES_REQUESTED** [review]

File: app/lib/ui/components/component_detail_screen.dart [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  const VariantEntry({required this.variant});
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      leading: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Icon(Icons.play_circle_outline),
```
Comment (paulinea) [inline app/lib/ui/components/component_detail_screen.dart:55]: This icon should be the same as in the Android app. Here it is a grey icon, on Android it is black.

**Review by paulinea — APPROVED** [review]

**Pull request merged**